src/utilities/routerUtilities.js

```diff
@@ -11,19 +9,31 @@ function handleServiceError(error, res) {
     return res.status(statusCode).json({ message });
 }
 
-function isValidBodyProperty(req, res, propertyName) {
-    const property = req.body[propertyName];
-
-    if (!isValidString(property)) {
-        res.status(400).json({
-            message: `Invalid property ${propertyName}`
-        });
-        return false;
+/*
+ summary: Returns a middleware function using specified params
+ parameters:
+    propertyName: The name of the property to check
+    isValidCallback: A callback function that takes the body param 
+        and returns a bool indicating if the value is valid 
+    required: If false, middleware will still pass even if property is not found
+*/
+function validateBody(propertyName, isValidCallback, required = true) {
```

Comment:
Maybe an object would reduce the amount of times this is called. 1 argument that takes in what we expect for each route. (If you decide to remove the middle functions earlier) Then we'd just loop through the object keys. It couldn't be a middleware though because we'd have to set up that object somewhere. Pros and cons, we'd have to weigh them against what you currently have

``` Javascript
const properties {
  propetyName: {callback: ()=> {}, required: true}
}
Object.keys(properties).forEach((property) => {"do stuff"})
```

Reply:
I like this idea, but I don't see there being much of a difference between repeatedly calling `validateBody` and calling it once, since it would still need to validate each property sequentially.

Reply:
No difference at all, it was just a different way to organize the code